### ISSUE TITLE
CannotFit error and anchors 

### ISSUE THREAD

**Issue by AlexanderZobkov** [asciidoctor/asciidoctor-pdf#447]
Anchors that are placed in table cells:
- affect on whether this or that element can fit a table 
- and can cause `Prawn::Errors::CannotFit` error. 

The reproducer and stacktrace are attached
[cannotfit-reproducer.txt](https://github.com/asciidoctor/asciidoctor-pdf/files/272613/cannotfit-reproducer.txt)
[cannotfit-stacktrace.txt](https://github.com/asciidoctor/asciidoctor-pdf/files/272614/cannotfit-stacktrace.txt)

**Comment by mojavelinux** [comment]
This looks like a failed assumption inside of Prawn. If the fragment is empty, as is the case for an inline anchor, it assumes that the fragment cannot fit. I'll need to find a way to trick Prawn into thinking that the fragment is not empty.

**Comment by mojavelinux** [comment]
We use a zero-width space character as placeholder text for an anchor point. I've submitted a patch to remove this placeholder text before the fragments are passed on to the line wrapper.

**Comment by mojavelinux** [comment]
Unfortunately, this fix breaks references to anchors. We're going to have to unfix it and get it addressed in Prawn. However, there is a workaround. Put the anchor at the end of the cell content instead of the start. As long as the cell doesn't begin with a non-breaking space, Prawn doesn't get confused.

For the record, the following method in Prawn is the culprit:

https://github.com/prawnpdf/prawn/blob/2.1.0/lib/prawn/text/formatted/line_wrap.rb#L89-L120

(specifically the call to fragment_finished)

**Comment by mojavelinux** [comment]
Another way to reproduce this error is to start a cell with a zero-width space in a different font (such as bold).

```
|===
|Aaaaa Bbbbb | **{zwsp}**Aaaaa_Bbbbb_Cc |Aaaaa_Bbbbb_Ccc |2251 |2050 |Aaaaa Bbbbb
|===
```

The previous patch doesn't fix this case, which is further justification to me that it needs to be addressed upstream.

**Comment by mojavelinux** [comment]
I reported this problem upstream as https://github.com/prawnpdf/prawn/issues/984 and submitted a patch.

**Comment by mojavelinux** [comment]
The underlying issue has been fixed upstream. Therefore, no additional changes will be needed here. We'll just wait on a new Prawn release.

**Comment by mojavelinux** [comment]
There have been several upstream releases, so this needs to be retested.

**Comment by mojavelinux** [comment]
I can confirm this is fixed as of Prawn 2.2.0. I'll also add a regression test to verify.
